main.go

```diff
@@ -714,12 +709,14 @@ func (tf *transformer) handleDirectives(comments []*ast.CommentGroup) {
 				continue
 			}
 			// This directive has two arguments: "go:linkname localName newName"
-			localName := fields[1]
 
-			// The local name must not be obfuscated.
-			obj := tf.pkg.Scope().Lookup(localName)
-			if obj != nil {
-				tf.ignoreObjects[obj] = true
+			if isPrivate(tf.pkg.Path()) {
+				// obfuscate the local name if the package is private.
+				curPkg, err := listPackage(tf.pkg.Path())
```

Comment:
I'm a bit confused. Can you not use the curPkg global directly? If this package is actually different, then calling this variable curPkg is also confusing :)

Reply:
Oops, forgot the `curPkg` global was available for me to use... it's been a hot minute since I worked on code in `main.go` 😅